Fix favorite faction display

Syldarion opened the pull request:
These changes include a fix to the favorite faction display and an update to the formatting of the changed file. I'll remove the latter if it isn't wanted, it was just to make working on it easier.

The favorite faction fix is explained in the second commit message.

Syldarion commented:
For sure, I'll take another stab at this then.

norraxx commented:
@Syldarion Maybe, we could talk about this code at discord. You would explain what you want to achieve, and I'd say what I don't like on this commit, what do you think?

Syldarion commented:
@norraxx Basically, just trying to have favorite faction percentage based off all games rather than those found first. I can see how just re-using a nested for loop is undesirable, seeing as it'll be going through the same data and processing it the same way twice.

norraxx commented:
@Syldarion Imagine how much load you have to do, to just compute that stuff. As I said, let's talk about that problem on discord https://discord.gg/VYX25M.
Write me there, i'm `dragonite` user there, maybe we can figure how to make it better. As @Brutus5000 said your solution will make very big load on servers, web page will start minutes, and maybe you won't have enough memory, and what you want is compute 4 numbers. :-)

norraxx commented on scripts/updateLeagues.js:
```diff
@@ -5,244 +5,260 @@ let fs = require('fs');
-                
+
+                if (leagueData.games == undefined) leagueData.games = [];
+                if (leagueData.playerData == undefined) leagueData.playerData = [];
```
I don't like conditions like that, use brackets, it's ok to have 3 lines instead of 1.

Syldarion commented:
@norraxx I spent some time trying to come up with a better solution, but only managed to find either longer runtime or wrong data.
I'd ask that you try the change yourself. Locally, running with this change is within a couple seconds of the runtime of what is live, but comes with accurate data.
Or I could close this and open an issue.

Syldarion commented:
As mentioned in the commit message of what I just added, an obvious solution had been staring me in the face all along. The for loop I added previously is gone, and performance seems in line with live.

norraxx commented:
I would to meet on discord.

norraxx commented on scripts/updateLeagues.js:
```diff
@@ -5,244 +5,251 @@ let fs = require('fs');
+                    const game = games[k];
+                    const timeElapsed = game.endTimestamp - game.startTimestamp;
+
+                    for (l in game.participants) {
```
use let

norraxx commented on scripts/updateLeagues.js:
```diff
@@ -5,244 +5,251 @@ let fs = require('fs');
+
+                        // Let's check if we don't have that player already
+                        let already = false;
+                        for (category in leagueData.playerData) {
```
-//-

norraxx commented on scripts/updateLeagues.js:
```diff
@@ -5,244 +5,251 @@ let fs = require('fs');
-                                        found = true;
+                }
+
+                let categories = {};
```
you don't need it
